packages/zql/src/ivm/view-apply-change.ts

```diff
@@ -278,7 +284,14 @@ export function applyChange(
           );
           assert(found, 'node does not exists');
           const oldEntry = view[pos];
-          view.splice(pos, 1);
+          const rc = must(refCountMap.get(oldEntry));
```

Comment:
I think this code is repeated for remove, so maybe a little helper can be factored out.

Reply:
Good catch